File: src/lib/types/journal.ts
import { WithBase } from "./base";

export type NewJournalEntry = {
  nestlingId: number;
  title: string;
  content: string;
  entryDate: string;
};

export type JournalEntry = WithBase<NewJournalEntry>;

export type NewJournalTemplate = {
  nestlingId: number;
  name: string;
  content: string;
};

export type JournalTemplate = WithBase<NewJournalTemplate>;

File: src-tauri/src/models/journal.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewJournalEntry {
    pub nestling_id: i64,
    pub title: String,
    pub content: String,
    pub entry_date: String,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JournalEntry {
    pub id: i64,
    pub nestling_id: i64,
    pub title: String,
    pub content: String,
    pub entry_date: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewJournalTemplate {
    pub nestling_id: i64,
    pub name: String,
    pub content: String,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JournalTemplate {
    pub id: i64,
    pub nestling_id: i64,
    pub name: String,
    pub content: String,
    pub created_at: String,
    pub updated_at: String,
}